mdp/utils/covariance.py

```diff
@@ -133,6 +135,146 @@ def fix(self, center=True):
+            and random variables along the columns.
+        :type x: numpy.ndarray
+
+        :param dt: Sequence of time increments between vectors. Must be
+            of length *x.shape[0]-1* in case of a single call. In case of
+            multiple calls with intended time dependence - as opposed to
+            computing the results as a weighted avarage of separate calls -
+            *dt* must be of length *x.shape[0]* starting with the second call.
+            The first element will be considered as the time difference
+            between the last element of *x* in the preceding call and the
+            first element of *x* in the current call. In case of not supplying
+            a *x.shape[0]*-th element in *dt*, time dependence will be dropped.
+            If *dt* omitted entirely, it will be considered to be one everywhere
+            whereas time dependence between calls is still dropped.
+        :type dt: numpy.ndarray
+        """
+        if self._cov_mtx is None:
+            self._init_internals(x)
+        # cast input
+        x = mdp.utils.refcast(x, self._dtype)
+
+        # account for the gap between training phases
+        if self.tphase > 0:
+            if dt is not None and dt.shape[0] == x.shape[0]:
+                self._avg += (self.xlast + x[0, :])*dt[0]/2.
+                self._cov_mtx += (numx.outer(self.xlast, self.xlast) +
+                                  numx.outer(x[0, :], x[0, :]))*dt[0]/2.
+                self._tlen += dt[0]
+
+        # keep last observation for multiple training phases
+        self.xlast = x[-1, :].copy()
+        self.dtlast = dt[-1].copy() if dt is not None else 1.
+
+        # make sure dt is defined and convenient to use
+        _dt = 1. if dt is None else dt[-x.shape[0]+1:, None]
+        sqdt = numx.sqrt(_dt)
+        # update the covariance matrix, the average and the number of
+        # observations
+        # the implementation is analogous to the evenly sampled case
+        # values might get big as it is only normalized in fix
+        xcpy = numx.multiply(x[1:, :], sqdt)
+        self._cov_mtx += mdp.utils.mult(xcpy.T, xcpy)/2.
+        numx.multiply(x[:-1, :], sqdt, out=xcpy)
+        self._cov_mtx += mdp.utils.mult(xcpy.T, xcpy)/2.
+
+        numx.multiply(x[:-1, :], _dt, out=xcpy)
+        self._avg += xcpy.sum(axis=0)/2.
+
+        numx.multiply(x[1:, :], _dt, out=xcpy)
+        self._avg += xcpy.sum(axis=0)/2.
+        # should the x be reset to their original value, as they are not passed as a copy?
+        self._tlen += _dt.sum() if dt is not None else x.shape[0]-1
+        self._steps += x.shape[0]
+        self.tphase += 1
+
+    def fix(self, center=True):
+        """Returns a triple containing the generalised
+        covariance matrix, the average and length of the
+        sequence (in time/ summed increments).
+
+        The covariance matrix is then reset to a zero-state.
```

Comment:
a zero-state -> zero

Reply:
This was a minor cosmetical suggestion. Keep it as it is. I don't really care.

Reply:
Maybe it was used to differentiate to the integer 0.

Reply:
Yes, maybe. `zero-state` also underlines the mutable nature of the object. Keep it like it was.